src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -139,8 +138,8 @@ object RegressionUtils {
   def getSampleAnnotations(vds: MatrixTable, annots: Array[String], ec: EvalContext): IndexedSeq[Array[Option[Double]]] = {
     val aQs = annots.map(parseExprAsDouble(_, ec))
 
-    vds.sampleIdsAndAnnotations.map { case (s, sa) =>
-      ec.setAll(s, sa)
+    vds.sampleAnnotations.map { sa =>
+      ec.set(1, sa)
```

Comment:
This ec still has an s field (defined in getPhenosCovCompleteSamples).  It should be removed and this should be ec.set(0, sa).